src/java/org/apache/cassandra/cql3/statements/schema/AlterTableStatement.java

```diff
@@ -751,24 +751,25 @@ public static class AlterConstraints extends AlterTableStatement
         @Override
         public KeyspaceMetadata apply(Epoch epoch, KeyspaceMetadata keyspace, TableMetadata table, ClusterMetadata metadata)
         {
-            TableMetadata.Builder tableBuilder = table.unbuild().epoch(epoch);
-
-            for (ColumnMetadata column : tableBuilder.columns())
+            for (ColumnMetadata column : table.columns())
             {
                 if (column.name == columnName)
                 {
-                    constraints.validate(column);
-                    column.setColumnConstraints(constraints);
```

Comment:
This was using a mutable object and causing problems when detecting metadata differences during alter table executions. By following a similar pattern to the one followed on column masks, we ensure that the changes are detected.